app/src/main/java/com/owncloud/android/operations/CreateShareWithShareeOperation.java

```diff
@@ -143,10 +143,17 @@ protected RemoteOperationResult run(OwnCloudClient client) {
 
             //once creating share link update other information
             UpdateShareInfoOperation updateShareInfoOperation = new UpdateShareInfoOperation(share, getStorageManager());
-            updateShareInfoOperation.setExpirationDateInMillis(expirationDateInMillis);
+            if (expirationDateInMillis > 0) {
+                updateShareInfoOperation.setExpirationDateInMillis(expirationDateInMillis);
+            }
             updateShareInfoOperation.setHideFileDownload(hideFileDownload);
             updateShareInfoOperation.setLabel(label);
 
+            //update the permission using update info api
```

Comment:
Once this upstream PR(https://github.com/nextcloud/android/pull/11568) is merged by NC need to rebase or remove this code.